### PR TITLE
Version verification

### PR DESCRIPTION
**Pull request by CerineBnsd**
## Description

- Change the logic of ``supported_server_version`` to support future server versions with minor changes
- Add an appropriate error message when trying to connect to old server versions that do not implement version verification.

## Related Issue

None

## Type of change

- [X] This change affects the client

## How Has This Been Tested?
- Unit tests locally with different versions numbers
- Tested with version 0.3.0 of the server
- Tested with version 0.1.0 of the server

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings

### CODE REVIEW

File: client/blindai/utils/errors.py [hunk 1]
```diff
@@ -8,6 +8,9 @@ def check_rpc_exception(rpc_error):
     elif rpc_error.code() == grpc.StatusCode.UNAVAILABLE:
         return f"Failed to connect to GRPC server: code={rpc_error.code()} message={rpc_error.details()}"
 
+    elif rpc_error.code() == grpc.StatusCode.UNIMPLEMENTED:
```
Comment (JoFrost) [inline client/blindai/utils/errors.py:11]: Why are you using this error code?

**Review by JoFrost — COMMENTED** [review]
Sounds good, I am however curious about the unimplemented error code.

File: client/blindai/utils/errors.py (hunk 1 above)
Comment (CerineBnsd) [reply client/blindai/utils/errors.py:11]: If we try to connect with the server 0.1.0 for example with this client, the ``stub.GetServerInfo`` call will raise a rgc_error with unimplemented code, as GetServerInfo is not implemented in that version on the server, and this exception will thus be raised whenever the client is calling an RPC procedure that is not implemented by the server, which means the versions are not compatible.

Comment (JoFrost) [reply client/blindai/utils/errors.py:11]: Cool. Clear.

**Review by JoFrost — APPROVED** [review]
All good.

**Pull request merged**